Yatima/Expr.lean

```diff
@@ -1,62 +1,86 @@
-import Yatima.Env
 import Yatima.Univ
 
 namespace Yatima
 
-  inductive Bind
-  | default
-  | implicit
-  | strictImplicit
-  | instImplicit
-  deriving BEq, Inhabited
-
-  inductive LitType
+inductive LitType
   | nat : LitType
   | str : LitType
   deriving BEq, Inhabited
 
-  inductive Literal
+inductive Literal
   | nat : Nat → Literal
   | str : String → Literal
   deriving BEq, Inhabited
 
-  inductive Expr
+inductive BinderInfo
+  | default
+  | implicit
+  | strictImplicit
+  | instImplicit
+  | auxDecl
+  deriving BEq, Inhabited
+
+inductive Expr
   | var   : Name → Nat → Expr
-  | sort  : Univ → Expr
-  | const : Name → ConstCid → List Univ → Expr
-  | app   : Expr → Expr → Expr
-  | lam   : Name → Bind → Expr → Expr → Expr
-  | pi    : Name → Bind → Expr → Expr → Expr
-  | letE  : Name → Expr → Expr → Expr → Expr
+  | sort  : UnivCid → Expr
+  | const : Name → ConstCid → List UnivCid → Expr
+  | app   : ExprCid → ExprCid → Expr
+  | lam   : Name → BinderInfo → ExprCid → ExprCid → Expr
+  | pi    : Name → BinderInfo → ExprCid → ExprCid → Expr
+  | letE  : Name → ExprCid → ExprCid → ExprCid → Expr
   | lit   : Literal → Expr
   | lty   : LitType → Expr
-  | fix   : Name → Expr → Expr
+  | fix   : Name → ExprCid → Expr
   deriving BEq, Inhabited
 
-  inductive ExprAnon
+inductive ExprAnon
   | var   : Nat → ExprAnon
   | sort  : UnivAnonCid → ExprAnon
   | const : ConstAnonCid → List UnivAnonCid → ExprAnon
   | app   : ExprAnonCid → ExprAnonCid → ExprAnon
-  | lam   : Bind → ExprAnonCid → ExprAnonCid → ExprAnon
-  | pi    : Bind → ExprAnonCid → ExprAnonCid → ExprAnon
+  | lam   : ExprAnonCid → ExprAnonCid → ExprAnon
+  | pi    : ExprAnonCid → ExprAnonCid → ExprAnon
   | letE  : ExprAnonCid → ExprAnonCid → ExprAnonCid → ExprAnon
   | lit   : Literal → ExprAnon
   | lty   : LitType → ExprAnon
   | fix   : ExprAnonCid → ExprAnon
   deriving BEq, Inhabited
 
-  inductive ExprMeta
+inductive ExprMeta
   | var   : Name → ExprMeta
   | sort  : UnivMetaCid → ExprMeta
   | const : Name → ConstMetaCid → List UnivMetaCid → ExprMeta
   | app   : ExprMetaCid → ExprMetaCid → ExprMeta
-  | lam   : Name → ExprMetaCid → ExprMetaCid → ExprMeta
-  | pi    : Name → ExprMetaCid → ExprMetaCid → ExprMeta
+  | lam   : Name → BinderInfo → ExprMetaCid → ExprMetaCid → ExprMeta
+  | pi    : Name → BinderInfo → ExprMetaCid → ExprMetaCid → ExprMeta
   | letE  : ExprMetaCid → ExprMetaCid → ExprMetaCid → ExprMeta
-  | lit   : ExprMeta
-  | lty   : ExprMeta
+  | lit   : Literal → ExprMeta
+  | lty   : LitType → ExprMeta
```

Comment:
Attention: I changed this!
Is this correct? Or should these constructors remain parameterless?

Reply:
On ExprMeta we shouldn't have the Literal/LitType values, no.